File: blog-service/src/main/java/mcb/blogs/publisher/BlogPost.java
package mcb.blogs.publisher;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import mcb.blogs.users.BlogUser;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
public class BlogPost {
    @Id
    @GeneratedValue
    private Long id;

    private String title;

    private String body;

    private long likes;


//    private BlogPost parentPost;

    @OneToOne(cascade = CascadeType.MERGE , orphanRemoval = true, fetch = FetchType.EAGER)
    private BlogUser creator;

    protected BlogPost() {
        this.likes=0;
        replies = new ArrayList<>();
    }

    public BlogPost(String title, String body,long userId) {
        this.title = title;
        this.body = body;
        this.likes=0;
        this.creator= new BlogUser();
        replies = new ArrayList<>();
        creator.setId(userId);

}

    public Long getId() {
        return id;
    }


    public Long getLikes() {
        return likes;
    }


    public Long setLikes(Long likes) {
        return this.likes = likes;
    }


    public String getTitle() {
        return title;
    }

    public String getBody() {
        return body;
    }



    public BlogUser getCreator() {
        return creator;
    }

    public void setCreator(BlogUser user) {
        this.creator=user;
    }

    @OneToMany(cascade = {CascadeType.PERSIST,CascadeType.DETACH,CascadeType.REMOVE,CascadeType.REFRESH}, orphanRemoval = true, fetch = FetchType.LAZY)
   // @JsonManagedReference
    List<BlogPost> replies ;

    public void setReplies(List<BlogPost> replies)
    {
        this.replies= replies;
    }

    public List<BlogPost> getReplies()
    {
       return replies;
    }

    public void addReply(BlogPost reply)
    {
        this.replies.add(reply);
    }

    public void addLike()
    {
        ++this.likes;
    }

    public void removeLike()
    {
        --this.likes;
    }

}

File: blog-ui/src/app/BlogPost.ts

export class BlogPost {
  private _id: number;
  private _title: string;
  private _body: string;
  private _likes: number;
  private _replies: BlogPost[];



  public constructor(id: number, title: string, body: string, likes: number) {
    this._title = title;
    this._body = body;
    this._likes = likes;
    this._id = id;
    this._replies = new Array();
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get title(): string {
    return this._title;
  }

  set title(value: string) {
    this._title = value;
  }

  get body(): string {
    return this._body;
  }

  set body(value: string) {
    this._body = value;
  }

  get likes(): number {
    return this._likes;
  }

  set likes(value: number) {
    this._likes = value;
  }

  get replies(): BlogPost[] {
    return this._replies;
  }

  set replies(value: BlogPost[]) {
    this._replies = value;
  }



}
